### PR TITLE
Update CITATION.cff

### PR DESCRIPTION
**Pull request by stefsmeets**
Hi all, this PR adds myself to the CITATION.cff. I also updated the version number and added CITATION.cff to the bumpversion config.

### CODE REVIEW

File: CITATION.cff [hunk 1]
```diff
@@ -14,6 +14,11 @@ authors:
   given-names: "Laurent"
   orcid: "https://orcid.org/0000-0002-5422-8794"
 
+- given-names: "Stef"
+  family-names: "Smeets"
+  affiliation: "Netherlands eScience Center"
+  orcid: "https://orcid.org/0000-0001-5107-3531"
+
 - family-names: "Saste"
```
Comment (timlichtenberg) [inline CITATION.cff:22]: There has been a typo here from before. This should read "Sastre".

**Review by timlichtenberg — APPROVED** [review]
Looks good, one legacy typo.

**Comment by stefsmeets** [comment]
Cheers, I also snuck a minor documentation fix that @lsoucasse just pointed out to me 😅

**Review by lsoucasse — APPROVED** [review]

**Comment by lsoucasse** [comment]
Approving this since Tim already gave a go.

**Pull request merged**